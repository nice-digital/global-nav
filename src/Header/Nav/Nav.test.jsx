import React from "react";
import Nav from "./Nav";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

import services from "./__mocks__/services.json";
import {
	eventName,
	defaultEventCategory,
	headerClickEventAction,
	eventTimeout,
} from "../../tracker";

jest.mock("./../../services.json", () => require("./__mocks__/services.json"));

describe("Nav", () => {
	const defaultProps = {
		isExpanded: false,
		accountsData: null,
		service: null,
	};
	const externalServices = services.external;

	it("Renders without crashing", () => {
		const wrapper = shallow(<Nav {...defaultProps} />);
		expect(wrapper).toHaveLength(1);
	});

	it("Matches snapshot with no accounts links", () => {
		const wrapper = shallow(<Nav {...defaultProps} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("Matches snapshot with single accounts link", () => {
		const accountsLinks = {
			"Sign in": "https://accounts.nice.org.uk/",
		};
		const wrapper = shallow(
			<Nav {...defaultProps} accountsLinks={accountsLinks} />
		);
		expect(toJson(wrapper.find("nav").at(1))).toMatchSnapshot();
	});

	it("Matches snapshot with multiple accounts links", () => {
		const accountsLinks = {
			"Joe Bloggs": "https://accounts.nice.org.uk/profile",
			"Sign out": "https://accounts.nice.org.uk/signout",
		};
		const wrapper = shallow(
			<Nav {...defaultProps} accountsLinks={accountsLinks} />
		);
		expect(toJson(wrapper.find("nav").at(1))).toMatchSnapshot();
	});

	it("Matches snapshot with sub links for selected external service", () => {
		const wrapper = mount(
			<Nav {...defaultProps} service={externalServices[1].id} />
		);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("Adds aria-current=true attribute for selected service", () => {
		const wrapper = shallow(
			<Nav {...defaultProps} service={externalServices[1].id} />
		);
		expect(wrapper.find("a").at(1).props()["aria-current"]).toEqual(true);
	});

	it("Adds aria-current=page attribute for selected service when matches current URL", () => {
		const oldLocation = global.window.location;
		delete global.window.location;
		global.window.location = new URL(externalServices[1].href);

		const wrapper = shallow(
			<Nav {...defaultProps} service={externalServices[1].id} />
		);
		expect(wrapper.find("a").at(1).props()["aria-current"]).toEqual("page");

		// Tidy up
		global.window.location = oldLocation;
	});

	it("Add expanded class to root element when passed isExpanded prop", () => {
		const wrapper = shallow(<Nav {...defaultProps} isExpanded={false} />);
		expect(wrapper.props().className).not.toContain("wrapperExpanded");
		wrapper.setProps({ isExpanded: true });
		expect(wrapper.props().className).toContain("wrapperExpanded");
	});

	describe("tracking", () => {
		beforeEach(() => {
			window.dataLayer = [];
		});

		afterAll(() => {
			// Cleanup
			delete window.dataLayer;
		});

		it("should push dataLayer event for nav item click", () => {
			const wrapper = shallow(<Nav {...defaultProps} isExpanded={false} />);

			wrapper.find("a[href='https://url1/']").simulate("click", {
				preventDefault: () => {},
				currentTarget: {
					// Mock e.currentTarget.getAttribute("href")
					getAttribute: () => "",
					textContent: "First link",
				},
			});

			expect(window.dataLayer).toEqual([
				{
					event: eventName,
					eventCategory: defaultEventCategory,
					eventAction: headerClickEventAction,
					eventLabel: "First link",
					eventCallback: expect.any(Function),
					eventTimeout: eventTimeout,
				},
			]);
		});

		it("should prevent default and navigate in event callback on nav item click", () => {
			const wrapper = shallow(<Nav {...defaultProps} isExpanded={false} />);

			const preventDefault = jest.fn();

			wrapper.find("a[href='https://url1/']").simulate("click", {
				preventDefault: preventDefault,
				currentTarget: {
					// Mock e.currentTarget.getAttribute("href")
					getAttribute: () => "https://url1/",
				},
			});

			expect(preventDefault).toHaveBeenCalled();

			window.dataLayer[0].eventCallback();
			expect(window.location.href).toEqual("https://url1/");
		});

		describe("Accounts links", () => {
			const accountsLinks = {
				"Joe Bloggs": "https://accounts.nice.org.uk/users/12345/editprofile",
				"Sign out": "https://accounts.nice.org.uk/signout",
				Admin: "https://accounts.nice.org.uk/admin",
			};

			it("should track edit profile link", () => {
				const preventDefault = jest.fn();
				const wrapper = shallow(
					<Nav
						{...defaultProps}
						isExpanded={true}
						accountsLinks={accountsLinks}
					/>
				);

				wrapper
					.find(
						"a[href='https://accounts.nice.org.uk/users/12345/editprofile']"
					)
					.simulate("click", {
						preventDefault: preventDefault,
						currentTarget: {
							// Mock e.currentTarget.getAttribute("href")
							getAttribute: () =>
								"https://accounts.nice.org.uk/users/12345/editprofile",
							textContent: "Joe Bloggs",
						},
					});

				expect(preventDefault).toHaveBeenCalled();
				expect(window.dataLayer).toEqual([
					{
						event: eventName,
						eventCategory: defaultEventCategory,
						eventAction: headerClickEventAction,
						eventLabel: "Edit profile",
						destinationUrl:
							"https://accounts.nice.org.uk/users/12345/editprofile",
						eventCallback: expect.any(Function),
						eventTimeout: eventTimeout,
					},
				]);
			});

			it("should track sign out link", () => {
				const preventDefault = jest.fn();
				const wrapper = shallow(
					<Nav
						{...defaultProps}
						isExpanded={true}
						accountsLinks={accountsLinks}
					/>
				);

				wrapper
					.find("a[href='https://accounts.nice.org.uk/signout']")
					.simulate("click", {
						preventDefault: preventDefault,
						currentTarget: {
							// Mock e.currentTarget.getAttribute("href")
							getAttribute: () => "https://accounts.nice.org.uk/signout",
							textContent: "Some log out text",
						},
					});

				expect(preventDefault).toHaveBeenCalled();
				expect(window.dataLayer).toEqual([
					{
						event: eventName,
						eventCategory: defaultEventCategory,
						eventAction: headerClickEventAction,
						eventLabel: "Sign out",
						destinationUrl: "https://accounts.nice.org.uk/signout",
						eventCallback: expect.any(Function),
						eventTimeout: eventTimeout,
					},
				]);
			});

			it("should not track admin link", () => {
				const preventDefault = jest.fn();
				const wrapper = shallow(
					<Nav
						{...defaultProps}
						isExpanded={true}
						accountsLinks={accountsLinks}
					/>
				);

				wrapper
					.find("a[href='https://accounts.nice.org.uk/admin']")
					.simulate("click", {
						preventDefault: preventDefault,
						currentTarget: {
							// Mock e.currentTarget.getAttribute("href")
							getAttribute: () => "https://accounts.nice.org.uk/admin",
							textContent: "Admin",
						},
					});

				expect(preventDefault).not.toHaveBeenCalled();
				expect(window.dataLayer).toEqual([]);
			});
		});
	});

	describe("internalServices", () => {
		const internalServices = services.internal;

		it("Matches snapshot with sub links for selected internal service", () => {
			const wrapper = mount(
				<Nav {...defaultProps} service={internalServices[0].id} />
			);
			expect(toJson(wrapper)).toMatchSnapshot();
		});

		it("Internal service 1 only renders itself and no other services", () => {
			const wrapper = mount(
				<Nav {...defaultProps} service={internalServices[0].id} />
			);
			const links = wrapper.find("a[href='/']");
			expect(links.length).toEqual(1);
			expect(links.text()).toEqual("First internal service");
		});

		it("Internal service 2 only renders itself and no other services", () => {
			const wrapper = mount(
				<Nav {...defaultProps} service={internalServices[1].id} />
			);
			const links = wrapper.find("a[href='/']");
			expect(links.length).toEqual(1);
			expect(links.text()).toEqual("Second internal service");
		});

		const propsWithAdditionalSubMenuItem = {
			...defaultProps,
			additionalSubMenuItems: [
				{
					service: internalServices[0].id,
					links: [{ text: "Admin", url: "/admin" }],
				},
			],
		};

		it("Internal service 1 adds additional sub menu item", () => {
			const wrapper = mount(
				<Nav
					{...propsWithAdditionalSubMenuItem}
					service={internalServices[0].id}
				/>
			);
			const links = wrapper.find("a[href='/admin']");
			expect(links.length).toEqual(1);
			expect(links.text()).toEqual("Admin");
		});

		it("Internal service 2 doesn't show the additional sub menu item for service 1", () => {
			const wrapper = mount(
				<Nav
					{...propsWithAdditionalSubMenuItem}
					service={internalServices[1].id}
				/>
			);
			const links = wrapper.find("a[href='/admin']");
			expect(links.length).toEqual(0);
		});

		const propsWithAdditionalSubMenuItemForEachService = {
			...defaultProps,
			additionalSubMenuItems: [
				{
					service: internalServices[0].id,
					links: [{ text: "Admin 1", url: "/admin1" }],
				},
				{
					service: internalServices[1].id,
					links: [{ text: "Admin 2", url: "/admin2" }],
				},
			],
		};

		it("Internal service 1 gets additional sub menu items when multiple are passed", () => {
			const wrapper = mount(
				<Nav
					{...propsWithAdditionalSubMenuItemForEachService}
					service={internalServices[0].id}
				/>
			);
			const links = wrapper.find("a[href='/admin1']");
			expect(links.length).toEqual(1);
			expect(links.text()).toEqual("Admin 1");
		});

		it("Internal service 2 gets additional sub menu items when multiple are passed", () => {
			const wrapper = mount(
				<Nav
					{...propsWithAdditionalSubMenuItemForEachService}
					service={internalServices[1].id}
				/>
			);
			const links = wrapper.find("a[href='/admin2']");
			expect(links.length).toEqual(1);
			expect(links.text()).toEqual("Admin 2");
		});
	});
});
