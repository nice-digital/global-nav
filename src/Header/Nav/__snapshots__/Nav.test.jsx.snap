// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Matches snapshot with multiple accounts links 1`] = `
<div>
  <div
    class="wrapper"
    id="header-menu"
  >
    <nav
      aria-label="primary navigation"
      class="nav"
      data-tracking="Primary navigation"
    >
      <div
        class="menuWrapper"
      >
        <ul
          aria-labelledby="header-menu-button"
          class="menuList"
        >
          <li
            data-tracking="First link"
          >
            <button
              aria-controls="dropdown-more-from-nice"
              aria-expanded="false"
              aria-label="First link"
              class="link"
              id="navlink-more-from-nice"
            >
              <span>
                First link
              </span>
              <svg
                aria-hidden="true"
                class="icon"
                fill="none"
                focusable="false"
                height="1em"
                pointer-events="none"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M478.152 174c0 10.091-3.525 18.661-10.576 25.712l-186 186c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864l-186-186C36.621 192.853 33 184.283 33 174c0-10.091 3.621-18.757 10.864-26L65 126.576C72.424 119.525 81.09 116 91 116c10.091 0 18.661 3.525 25.712 10.576L255.576 265.44 394.44 126.576c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576L467.576 148c7.051 7.424 10.576 16.091 10.576 26Z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <div
              class="dropdown wrapper"
              data-tracking="First link dropdown"
              id="dropdown-more-from-nice"
            >
              <div
                class="container"
              >
                <a
                  class="skiplink"
                  href="#navlink-link2"
                >
                  Skip 
                  First link
                   submenu
                </a>
                <section
                  aria-label="More from NICE - navigation"
                >
                  <h2
                    class="mt--0"
                  >
                    Explore more from NICE
                  </h2>
                  <div
                    class="grid"
                    gutter="loose"
                  >
                    <div
                      class="gridItem aboutCol"
                      data-tracking="Explore more"
                    >
                      <ul
                        class="listColumnar mt--d"
                      >
                        <li>
                          <a
                            href="https://www.test.nice.org/what-nice-does"
                          >
                            What NICE does
                          </a>
                        </li>
                        <li>
                          <a
                            href="https://www.test.nice.org/about-us"
                          >
                            About us
                          </a>
                        </li>
                        <li>
                          <a
                            href="https://www.test.nice.org/reusing-our-content"
                          >
                            Reusing our content
                          </a>
                        </li>
                        <li>
                          <a
                            href="https://www.test.nice.org/implementing-nice-guidance"
                          >
                            Implementing NICE guidance
                          </a>
                        </li>
                        <li>
                          <a
                            href="https://www.test.nice.org/news"
                          >
                            News, blogs and podcasts
                          </a>
                        </li>
                        <li>
                          <a
                            href="https://www.test.nice.org/contact-us"
                          >
                            Contact us
                          </a>
                        </li>
                        <li>
                          <a
                            href="https://www.test.nice.org/get-involved"
                          >
                            Get involved
                          </a>
                        </li>
                        <li>
                          <a
                            href="https://www.test.nice.org/careers"
                          >
                            Careers
                          </a>
                        </li>
                      </ul>
                    </div>
                  </div>
                  <hr />
                  <ul
                    class="inlineList mt--d"
                    data-tracking="Quick links"
                  >
                    <li>
                      <a
                        href="https://www.test.nice.org/get-involved/our-committees"
                      >
                        Join a committee
                      </a>
                    </li>
                    <li>
                      <a
                        href="https://www.test.nice.org/get-involved/consultations"
                      >
                        Comment on a consultation
                      </a>
                    </li>
                  </ul>
                </section>
                <button
                  class="exit"
                  data-tracking="Close menu"
                >
                  Close menu 
                  <svg
                    aria-hidden="true"
                    class="icon"
                    fill="none"
                    focusable="false"
                    height="1em"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M480 392.508c0 10.052-3.52 18.598-10.559 25.637l-51.296 51.296C411.106 476.48 402.56 480 392.508 480c-10.053 0-18.598-3.52-25.638-10.559L256 358.571l-110.87 110.87C138.09 476.48 129.545 480 119.492 480c-10.052 0-18.598-3.52-25.637-10.559l-51.296-51.296C35.52 411.106 32 402.56 32 392.508c0-10.053 3.52-18.598 10.56-25.638L153.428 256 42.559 145.13C35.52 138.09 32 129.545 32 119.492c0-10.052 3.52-18.598 10.56-25.637l51.295-51.296C100.894 35.52 109.44 32 119.492 32c10.053 0 18.598 3.52 25.638 10.56L256 153.428l110.87-110.87C373.91 35.52 382.455 32 392.508 32c10.052 0 18.598 3.52 25.637 10.56l51.296 51.295C476.48 100.894 480 109.44 480 119.492c0 10.053-3.52 18.598-10.559 25.638L358.571 256l110.87 110.87C476.48 373.91 480 382.455 480 392.508Z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </li>
          <li
            data-tracking="Second link"
          >
            <a
              aria-label="Second link abbreviation title (lnk2)"
              class="link"
              href="https://www.test-link2.nice.org/url2/"
              id="navlink-link2"
            >
              <span>
                Second link
              </span>
            </a>
          </li>
          <li
            data-tracking="Third link"
          >
            <a
              aria-label="Third link abbreviation title (lnk3)"
              class="link"
              href="https://www.test.nice.org/url3/"
              id="navlink-link3"
            >
              <span>
                Third link
              </span>
              <svg
                aria-hidden="true"
                class="icon"
                fill="none"
                focusable="false"
                height="1em"
                pointer-events="none"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M478.152 174c0 10.091-3.525 18.661-10.576 25.712l-186 186c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864l-186-186C36.621 192.853 33 184.283 33 174c0-10.091 3.621-18.757 10.864-26L65 126.576C72.424 119.525 81.09 116 91 116c10.091 0 18.661 3.525 25.712 10.576L255.576 265.44 394.44 126.576c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576L467.576 148c7.051 7.424 10.576 16.091 10.576 26Z"
                  fill="currentColor"
                />
              </svg>
            </a>
            <ul
              aria-label="More NICE services"
              class="nonJsDropdown"
              id="link3"
            >
              <li>
                <a
                  href="https://nestedLink1"
                >
                  Nested link one
                </a>
              </li>
              <li>
                <a
                  href="https://nestedLink2"
                >
                  Nested link two
                </a>
              </li>
            </ul>
          </li>
          <li
            data-tracking="Fourth link"
          >
            <a
              aria-label="Fourth link"
              class="link"
              href="https://www.test.nice.org/url4/"
              id="navlink-test-service"
            >
              <span>
                Fourth link
              </span>
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <nav
      aria-label="My account"
      class="nav myAccount"
    >
      <h2
        class="myAccountHeading"
      >
        My account
      </h2>
      <div
        class="menuWrapper"
      >
        <ul
          class="menuList"
        >
          <li>
            <a
              class="link"
              href="https://accounts.nice.org.uk/profile"
            >
              Joe Bloggs
            </a>
          </li>
          <li>
            <a
              class="link"
              href="https://accounts.nice.org.uk/signout"
            >
              Sign out
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</div>
`;

exports[`Nav Matches snapshot with no accounts links 1`] = `
<div>
  <div
    class="wrapper"
    id="header-menu"
  >
    <nav
      aria-label="primary navigation"
      class="nav"
      data-tracking="Primary navigation"
    >
      <div
        class="menuWrapper"
      >
        <ul
          aria-labelledby="header-menu-button"
          class="menuList"
        >
          <li
            data-tracking="First link"
          >
            <button
              aria-controls="dropdown-more-from-nice"
              aria-expanded="false"
              aria-label="First link"
              class="link"
              id="navlink-more-from-nice"
            >
              <span>
                First link
              </span>
              <svg
                aria-hidden="true"
                class="icon"
                fill="none"
                focusable="false"
                height="1em"
                pointer-events="none"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M478.152 174c0 10.091-3.525 18.661-10.576 25.712l-186 186c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864l-186-186C36.621 192.853 33 184.283 33 174c0-10.091 3.621-18.757 10.864-26L65 126.576C72.424 119.525 81.09 116 91 116c10.091 0 18.661 3.525 25.712 10.576L255.576 265.44 394.44 126.576c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576L467.576 148c7.051 7.424 10.576 16.091 10.576 26Z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <div
              class="dropdown wrapper"
              data-tracking="First link dropdown"
              id="dropdown-more-from-nice"
            >
              <div
                class="container"
              >
                <a
                  class="skiplink"
                  href="#navlink-link2"
                >
                  Skip 
                  First link
                   submenu
                </a>
                <section
                  aria-label="More from NICE - navigation"
                >
                  <h2
                    class="mt--0"
                  >
                    Explore more from NICE
                  </h2>
                  <div
                    class="grid"
                    gutter="loose"
                  >
                    <div
                      class="gridItem aboutCol"
                      data-tracking="Explore more"
                    >
                      <ul
                        class="listColumnar mt--d"
                      >
                        <li>
                          <a
                            href="https://www.test.nice.org/what-nice-does"
                          >
                            What NICE does
                          </a>
                        </li>
                        <li>
                          <a
                            href="https://www.test.nice.org/about-us"
                          >
                            About us
                          </a>
                        </li>
                        <li>
                          <a
                            href="https://www.test.nice.org/reusing-our-content"
                          >
                            Reusing our content
                          </a>
                        </li>
                        <li>
                          <a
                            href="https://www.test.nice.org/implementing-nice-guidance"
                          >
                            Implementing NICE guidance
                          </a>
                        </li>
                        <li>
                          <a
                            href="https://www.test.nice.org/news"
                          >
                            News, blogs and podcasts
                          </a>
                        </li>
                        <li>
                          <a
                            href="https://www.test.nice.org/contact-us"
                          >
                            Contact us
                          </a>
                        </li>
                        <li>
                          <a
                            href="https://www.test.nice.org/get-involved"
                          >
                            Get involved
                          </a>
                        </li>
                        <li>
                          <a
                            href="https://www.test.nice.org/careers"
                          >
                            Careers
                          </a>
                        </li>
                      </ul>
                    </div>
                  </div>
                  <hr />
                  <ul
                    class="inlineList mt--d"
                    data-tracking="Quick links"
                  >
                    <li>
                      <a
                        href="https://www.test.nice.org/get-involved/our-committees"
                      >
                        Join a committee
                      </a>
                    </li>
                    <li>
                      <a
                        href="https://www.test.nice.org/get-involved/consultations"
                      >
                        Comment on a consultation
                      </a>
                    </li>
                  </ul>
                </section>
                <button
                  class="exit"
                  data-tracking="Close menu"
                >
                  Close menu 
                  <svg
                    aria-hidden="true"
                    class="icon"
                    fill="none"
                    focusable="false"
                    height="1em"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M480 392.508c0 10.052-3.52 18.598-10.559 25.637l-51.296 51.296C411.106 476.48 402.56 480 392.508 480c-10.053 0-18.598-3.52-25.638-10.559L256 358.571l-110.87 110.87C138.09 476.48 129.545 480 119.492 480c-10.052 0-18.598-3.52-25.637-10.559l-51.296-51.296C35.52 411.106 32 402.56 32 392.508c0-10.053 3.52-18.598 10.56-25.638L153.428 256 42.559 145.13C35.52 138.09 32 129.545 32 119.492c0-10.052 3.52-18.598 10.56-25.637l51.295-51.296C100.894 35.52 109.44 32 119.492 32c10.053 0 18.598 3.52 25.638 10.56L256 153.428l110.87-110.87C373.91 35.52 382.455 32 392.508 32c10.052 0 18.598 3.52 25.637 10.56l51.296 51.295C476.48 100.894 480 109.44 480 119.492c0 10.053-3.52 18.598-10.559 25.638L358.571 256l110.87 110.87C476.48 373.91 480 382.455 480 392.508Z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </li>
          <li
            data-tracking="Second link"
          >
            <a
              aria-label="Second link abbreviation title (lnk2)"
              class="link"
              href="https://www.test-link2.nice.org/url2/"
              id="navlink-link2"
            >
              <span>
                Second link
              </span>
            </a>
          </li>
          <li
            data-tracking="Third link"
          >
            <a
              aria-label="Third link abbreviation title (lnk3)"
              class="link"
              href="https://www.test.nice.org/url3/"
              id="navlink-link3"
            >
              <span>
                Third link
              </span>
              <svg
                aria-hidden="true"
                class="icon"
                fill="none"
                focusable="false"
                height="1em"
                pointer-events="none"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M478.152 174c0 10.091-3.525 18.661-10.576 25.712l-186 186c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864l-186-186C36.621 192.853 33 184.283 33 174c0-10.091 3.621-18.757 10.864-26L65 126.576C72.424 119.525 81.09 116 91 116c10.091 0 18.661 3.525 25.712 10.576L255.576 265.44 394.44 126.576c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576L467.576 148c7.051 7.424 10.576 16.091 10.576 26Z"
                  fill="currentColor"
                />
              </svg>
            </a>
            <ul
              aria-label="More NICE services"
              class="nonJsDropdown"
              id="link3"
            >
              <li>
                <a
                  href="https://nestedLink1"
                >
                  Nested link one
                </a>
              </li>
              <li>
                <a
                  href="https://nestedLink2"
                >
                  Nested link two
                </a>
              </li>
            </ul>
          </li>
          <li
            data-tracking="Fourth link"
          >
            <a
              aria-label="Fourth link"
              class="link"
              href="https://www.test.nice.org/url4/"
              id="navlink-test-service"
            >
              <span>
                Fourth link
              </span>
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</div>
`;

exports[`Nav Matches snapshot with single accounts link 1`] = `
<div>
  <div
    class="wrapper"
    id="header-menu"
  >
    <nav
      aria-label="primary navigation"
      class="nav"
      data-tracking="Primary navigation"
    >
      <div
        class="menuWrapper"
      >
        <ul
          aria-labelledby="header-menu-button"
          class="menuList"
        >
          <li
            data-tracking="First link"
          >
            <button
              aria-controls="dropdown-more-from-nice"
              aria-expanded="false"
              aria-label="First link"
              class="link"
              id="navlink-more-from-nice"
            >
              <span>
                First link
              </span>
              <svg
                aria-hidden="true"
                class="icon"
                fill="none"
                focusable="false"
                height="1em"
                pointer-events="none"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M478.152 174c0 10.091-3.525 18.661-10.576 25.712l-186 186c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864l-186-186C36.621 192.853 33 184.283 33 174c0-10.091 3.621-18.757 10.864-26L65 126.576C72.424 119.525 81.09 116 91 116c10.091 0 18.661 3.525 25.712 10.576L255.576 265.44 394.44 126.576c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576L467.576 148c7.051 7.424 10.576 16.091 10.576 26Z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <div
              class="dropdown wrapper"
              data-tracking="First link dropdown"
              id="dropdown-more-from-nice"
            >
              <div
                class="container"
              >
                <a
                  class="skiplink"
                  href="#navlink-link2"
                >
                  Skip 
                  First link
                   submenu
                </a>
                <section
                  aria-label="More from NICE - navigation"
                >
                  <h2
                    class="mt--0"
                  >
                    Explore more from NICE
                  </h2>
                  <div
                    class="grid"
                    gutter="loose"
                  >
                    <div
                      class="gridItem aboutCol"
                      data-tracking="Explore more"
                    >
                      <ul
                        class="listColumnar mt--d"
                      >
                        <li>
                          <a
                            href="https://www.test.nice.org/what-nice-does"
                          >
                            What NICE does
                          </a>
                        </li>
                        <li>
                          <a
                            href="https://www.test.nice.org/about-us"
                          >
                            About us
                          </a>
                        </li>
                        <li>
                          <a
                            href="https://www.test.nice.org/reusing-our-content"
                          >
                            Reusing our content
                          </a>
                        </li>
                        <li>
                          <a
                            href="https://www.test.nice.org/implementing-nice-guidance"
                          >
                            Implementing NICE guidance
                          </a>
                        </li>
                        <li>
                          <a
                            href="https://www.test.nice.org/news"
                          >
                            News, blogs and podcasts
                          </a>
                        </li>
                        <li>
                          <a
                            href="https://www.test.nice.org/contact-us"
                          >
                            Contact us
                          </a>
                        </li>
                        <li>
                          <a
                            href="https://www.test.nice.org/get-involved"
                          >
                            Get involved
                          </a>
                        </li>
                        <li>
                          <a
                            href="https://www.test.nice.org/careers"
                          >
                            Careers
                          </a>
                        </li>
                      </ul>
                    </div>
                  </div>
                  <hr />
                  <ul
                    class="inlineList mt--d"
                    data-tracking="Quick links"
                  >
                    <li>
                      <a
                        href="https://www.test.nice.org/get-involved/our-committees"
                      >
                        Join a committee
                      </a>
                    </li>
                    <li>
                      <a
                        href="https://www.test.nice.org/get-involved/consultations"
                      >
                        Comment on a consultation
                      </a>
                    </li>
                  </ul>
                </section>
                <button
                  class="exit"
                  data-tracking="Close menu"
                >
                  Close menu 
                  <svg
                    aria-hidden="true"
                    class="icon"
                    fill="none"
                    focusable="false"
                    height="1em"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M480 392.508c0 10.052-3.52 18.598-10.559 25.637l-51.296 51.296C411.106 476.48 402.56 480 392.508 480c-10.053 0-18.598-3.52-25.638-10.559L256 358.571l-110.87 110.87C138.09 476.48 129.545 480 119.492 480c-10.052 0-18.598-3.52-25.637-10.559l-51.296-51.296C35.52 411.106 32 402.56 32 392.508c0-10.053 3.52-18.598 10.56-25.638L153.428 256 42.559 145.13C35.52 138.09 32 129.545 32 119.492c0-10.052 3.52-18.598 10.56-25.637l51.295-51.296C100.894 35.52 109.44 32 119.492 32c10.053 0 18.598 3.52 25.638 10.56L256 153.428l110.87-110.87C373.91 35.52 382.455 32 392.508 32c10.052 0 18.598 3.52 25.637 10.56l51.296 51.295C476.48 100.894 480 109.44 480 119.492c0 10.053-3.52 18.598-10.559 25.638L358.571 256l110.87 110.87C476.48 373.91 480 382.455 480 392.508Z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </li>
          <li
            data-tracking="Second link"
          >
            <a
              aria-label="Second link abbreviation title (lnk2)"
              class="link"
              href="https://www.test-link2.nice.org/url2/"
              id="navlink-link2"
            >
              <span>
                Second link
              </span>
            </a>
          </li>
          <li
            data-tracking="Third link"
          >
            <a
              aria-label="Third link abbreviation title (lnk3)"
              class="link"
              href="https://www.test.nice.org/url3/"
              id="navlink-link3"
            >
              <span>
                Third link
              </span>
              <svg
                aria-hidden="true"
                class="icon"
                fill="none"
                focusable="false"
                height="1em"
                pointer-events="none"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M478.152 174c0 10.091-3.525 18.661-10.576 25.712l-186 186c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864l-186-186C36.621 192.853 33 184.283 33 174c0-10.091 3.621-18.757 10.864-26L65 126.576C72.424 119.525 81.09 116 91 116c10.091 0 18.661 3.525 25.712 10.576L255.576 265.44 394.44 126.576c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576L467.576 148c7.051 7.424 10.576 16.091 10.576 26Z"
                  fill="currentColor"
                />
              </svg>
            </a>
            <ul
              aria-label="More NICE services"
              class="nonJsDropdown"
              id="link3"
            >
              <li>
                <a
                  href="https://nestedLink1"
                >
                  Nested link one
                </a>
              </li>
              <li>
                <a
                  href="https://nestedLink2"
                >
                  Nested link two
                </a>
              </li>
            </ul>
          </li>
          <li
            data-tracking="Fourth link"
          >
            <a
              aria-label="Fourth link"
              class="link"
              href="https://www.test.nice.org/url4/"
              id="navlink-test-service"
            >
              <span>
                Fourth link
              </span>
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <nav
      aria-label="My account"
      class="nav myAccount"
    >
      <div
        class="menuWrapper"
      >
        <ul
          class="menuList"
        >
          <li>
            <a
              class="link"
              href="https://accounts.nice.org.uk/"
            >
              Sign in
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</div>
`;

exports[`Nav internalServices Matches snapshot with sub links for selected internal service 1`] = `
<div>
  <div
    class="wrapper wrapperWithSubLinks"
    id="header-menu"
  >
    <nav
      aria-label="primary navigation"
      class="nav"
      data-tracking="Primary navigation"
    >
      <div
        class="menuWrapper"
      >
        <ul
          aria-labelledby="header-menu-button"
          class="menuList"
        >
          <li
            data-tracking="First internal service"
          >
            <a
              aria-current="true"
              aria-label="First internal service"
              class="link"
              href="https://www.test.nice.org/first-internal-service"
              id="navlink-internal1"
            >
              <span>
                First internal service
              </span>
            </a>
            <div
              class="wrapper"
            >
              <ul
                aria-label="First internal service links"
                class="list"
              >
                <li>
                  <a
                    aria-label="internal1 sub link 1"
                    class="link"
                    href="/i1sublink1"
                  >
                    internal1 sub link 1
                  </a>
                </li>
                <li>
                  <a
                    aria-label="internal1 sub link 2"
                    class="link"
                    href="/i2sublink2"
                  >
                    internal1 sub link 2
                  </a>
                </li>
              </ul>
            </div>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</div>
`;
